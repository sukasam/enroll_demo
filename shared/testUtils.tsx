import { ThemeProvider } from "@mui/material/styles";
import { render, RenderOptions } from "@testing-library/react";
import { lightTheme } from "Styles/theme/themes";

const customRender = (
    ui: React.ReactElement,
    options?: Omit<RenderOptions, "wrapper">
): ReturnType<typeof render> =>
    render(ui, {
        wrapper: ({ children }) => (
            <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
        ),
        ...options
    });

export * from "@testing-library/react";
export { customRender as render };
