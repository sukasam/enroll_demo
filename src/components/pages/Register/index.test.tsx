import { useRouter } from "next/router";
import { render } from "Shared/testUtils";
import RegisterPage from "./index";

// Mock external dependencies and hooks
jest.mock("Contexts/UserContext", () => ({
    useUser: (): { isLoggedIn: boolean; activeAccordionSection: number } => ({
        isLoggedIn: false,
        activeAccordionSection: 1
    })
}));
jest.mock(
    "Services/mixpanel/initializeMixPanel",
    (): {
        trackPageView: jest.Mock<() => void>;
    } => ({
        trackPageView: jest.fn()
    })
);
jest.mock("Constants/countryConfig", () => ({
    useCountryConfig: (): {
        isNoPurchaseMarket: boolean;
        marketExceptions: { hiddenShippingProvince: boolean };
    } => ({
        isNoPurchaseMarket: false,
        marketExceptions: { hiddenShippingProvince: false }
    })
}));
jest.mock("next/router", () => ({
    useRouter: jest.fn()
}));

// Mock custom hooks
jest.mock("./components/Catalog/hooks", () => ({
    __esModule: true,
    default: jest.fn()
}));
jest.mock("./hooks", () => ({
    useSectionsConfig: jest.fn(() => []),
    SectionFactory: jest.fn(() => null)
}));

describe("RegisterPage", () => {
    beforeEach(() => {
        // Setup router mock for each test
        (useRouter as jest.Mock).mockImplementation(() => ({
            replace: jest.fn(),
            prefetch: jest.fn(),
            pathname: "/register"
        }));
    });

    it("renders correctly", () => {
        const { container } = render(<RegisterPage />);
        expect(container).toMatchSnapshot();
    });
});
