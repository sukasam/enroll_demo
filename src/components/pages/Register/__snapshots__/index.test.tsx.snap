// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPage renders correctly 1`] = `
<div>
  <div
    class=MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-XXXXX-MuiGrid-root
  >
    <div
      class=MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 css-XXXXX-MuiGrid-root
    >
      <div
        class="css-xpzruu-accordionContainer"
        data-testid="accordion_container"
      />
    </div>
    <div
      class=MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 aside-wrapper css-XXXXX-MuiGrid-root
    />
  </div>
</div>
`;
